internal/backends/clojure/deps.go

```diff
@@ -0,0 +1,504 @@
+					util.Die("invalid \"deps.edn\", aborting")
+				}
+			}
+			return pedn
+		}
+	} else {
+		util.Die("error while reading \"deps.edn\", aborting")
+	}
+	panic("fglng")
+}
+
+func WriteDeps(ednMap interface{}) error {
+	bs, err := edn.MarshalIndent(ednMap, "", "  ")
+	if err != nil {
+		return err
+	} else {
+		ioutil.WriteFile("deps.edn", bs, 0644)
+		return nil
+	}
+	panic("fglng")
+}
+
+func addPackages(m map[api.PkgName]api.PkgSpec) {
+	pedn := ReadDeps()
+	var jpkg JavaPackage
+	for name, spec := range m {
+		err := jpkg.FromString(string(name))
+		if err != nil {
+			util.Die(err.Error())
+		}
+		mPkgInfo, err := GetMPackageInfo(jpkg)
+		if err != nil {
+			util.Die(err.Error())
+		}
+		version := string(spec)
+		if version == "" {
+			util.Log("Using version: " + mPkgInfo.LatestVersion + " for package: " + string(name))
+			m[name] = api.PkgSpec(mPkgInfo.LatestVersion)
+		} else {
+			versionValid := false
+			for _, v := range mPkgInfo.Versions {
+				if v == version {
+					versionValid = true
+				}
+			}
+			if ! versionValid {
+				util.Die("No artifact with version: \"" + string(spec) + "\" found for package: \"" + string(name) + "\"")
+			}
+		}
+	}
+
+	depsMap := pedn[edn.Keyword("deps")].(map[interface{}]interface{})
+	for name, spec := range m {
+		depsMap[edn.Symbol(string(name))] = map[edn.Keyword]string {
+			edn.Keyword("mvn/version"): string(spec),
+		}
+	}
+
+	WriteDeps(pedn)
+}
+
+func removePackages(m map[api.PkgName]bool) {
+	pedn := ReadDeps()
+	var jpkg JavaPackage
+
+	for name, _ := range m {
+		err := jpkg.FromString(string(name))
+		if err != nil {
+			util.Die(err.Error())
+		}
+	}
+
+	depsMap := pedn[edn.Keyword("deps")].(map[interface{}]interface{})
+	for name, _ := range m {
+		sname := string(name)
+		delete(depsMap, edn.Symbol(sname))
+	}
+
+	WriteDeps(pedn)
+}
+
+func listSpecfile() map[api.PkgName]api.PkgSpec {
+	pedn := ReadDeps()
+	depsMap := pedn[edn.Keyword("deps")].(map[interface{}]interface{})
+
+	res := map[api.PkgName]api.PkgSpec{}
+	for k, v := range depsMap {
+		name := k.(edn.Symbol).String()
+		t := v.(map[interface{}]interface{})
+		spec := t[edn.Keyword("mvn/version")].(string)
+		res[api.PkgName(name)] = api.PkgSpec(spec)
+	}
+	return res
+}
+
+func listLockfile() map[api.PkgName]api.PkgVersion {
+	traceEdn := struct {
+		Log []struct {
+			Lib edn.Symbol `edn:"lib"`
+			UsedVersion struct {
+				Version string `edn:"mvn/version"`
+			} `edn:"use-coord"`
+		} `edn:"log"`
+	} {}
+	bs, err := ioutil.ReadFile("trace.edn")
+	if err != nil {
+		util.Die("Error while reading trace.edn", err)
+	}
+	err = edn.Unmarshal(bs, &traceEdn)
+
+	res := map[api.PkgName]api.PkgVersion{}
+
+	for _, entry := range traceEdn.Log {
+		res[api.PkgName(string(entry.Lib))] = api.PkgVersion(entry.UsedVersion.Version)
+	}
+
+	return res
+}
+
+func install() {
+	util.Log("installing, this will download the dependencies")
+	util.RunCmd([]string {"clj", "--eval", "(System/exit 0)"})
```

Comment:
I'm not terribly familiar with clojure - will this command automatically run `src/main.clj`? What if a project has `./main.clj` and `src/main.clj`?

Reply:
`clj --eval "(System/exit 0)"` will download the dependencies, generate the classpath and launch an repl with that classpath which will exit immediately. It only needs `deps.edn` to work.